### PR TITLE
 Fix: ui_cluster: Improve the process of 'crm cluster stop' (bsc#1213889)

### PR DESCRIPTION
**Pull request by liangxin1300**
- Fix bsc#1213889: don't try to get DC when local service is unavailable
- Split into small functions to make the whole process clear
- Support checking and setting dlm option on remote node

### CODE REVIEW

File: crmsh/ui_cluster.py [hunk 1]
```diff
@@ -190,25 +190,36 @@
...
-                    logger.info("The cluster stack stopped on {}".format(node))
-                else:
-                    logger.info("The cluster stack already stopped on {}".format(node))
-                node_list.remove(node)
-            elif not utils.service_is_active("pacemaker.service", remote_addr=node):
+    @staticmethod
+    def _stop_partial_service(node):
+        """
+        Stops partial cluster service
+        """
```
Comment (nicholasyang2022) [inline crmsh/ui_cluster.py:197]: Can you rename this function or change the docstring to make the meaning of return value more clear?

File: crmsh/ui_cluster.py [hunk 2]
```diff
@@ -190,25 +190,36 @@ def do_start(self, context, *args):
...
+
+        return True
+
+    @staticmethod
+    def _wait_for_dc():
+        """
+        Wait for the cluster's DC to become available
+        """
+        if not utils.service_is_active("pacemaker.service"):
             return
```
Comment (nicholasyang2022) [inline crmsh/ui_cluster.py:223]: More comments are needed here to document why DC is waited for only when pacemaker.service is active.

**Review by nicholasyang2022 — CHANGES_REQUESTED** [review]

File: crmsh/ui_cluster.py [hunk 3]
```diff
@@ -190,25 +190,40 @@ def do_start(self, context, *args):
...
+            return False
+
+        return True
+
+    @staticmethod
+    def _wait_for_dc():
+        """
+        Wait for the cluster's DC to become available
+        """
+        # Cannot get DC when local service is not available
```
Comment (nicholasyang2022) [inline crmsh/ui_cluster.py:225]: Why can we ignore DC when local service is not started? What if DC is running on another node?

Comment (liangxin1300) [reply crmsh/ui_cluster.py:225]: Yeah, better to consider the remote case

**Pull request closed without merging**